### DISCUSSION TITLE
Dynamic Bargaining System - Ecommerce

### DISCUSSION THREAD

**Discussion by TAbhay** [scaleracademy/open-source-projects#65] [Ideas]
**Dynamic Bargaining System - Ecommerce**
**Negotiate & Buy**

Today there are a lot of E-Commerce platforms in the market but almost 99% of them are lacking a bargaining feature.
Suppose, Customer X loves buying from E-Store but the same product is available at local stores or some other platforms at the same pricing. Customer X can easily bargain the product at the local stores.  Wait !!

*Now, let's say E-Store is having a bargaining system, Customer X need can easily bargain and buy on E-Store.*

![Screenshot 2021-06-06 184513](https://user-images.githubusercontent.com/56065895/120930116-9d24cd80-c709-11eb-9c6e-c1ce2df98020.png)

The E-Store will be having the basic features and structures of an E-Commerce platform. The Hierarchical structure will be similar to as shown in fig.

![unnamed](https://user-images.githubusercontent.com/56065895/120930184-e6751d00-c709-11eb-9b5a-5664e5bb7d4a.png)

**METHODOLOGY**

_Instead of a Bargain amount, let’s say it is a bid_

- Every customer can bid on products that are eligible for a bid. Every product has a minimum bid amount above which bargaining is possible ( bidding Range ).

- This bidding system will have the feature to bid on a product. But a one-to-one bidding system can also be built as I mentioned it in Add On.

- Bidding will take place for a certain period of time after which results can be declared manually or we can use something like Vickrey Auction theory etc. for results.

- Customers can also see the number of current bids and the daily bidding frequencies on a product.

Summarizing

Product to Bid ->  Customers -> Certain Time Frame ( Bids ) -> Bid results 

**Working of Bidding Portal** 

1)  Eligibility for Bidding
       Listing of All products
          Product X 
              Name ..price .. CanBeBid ..MinimumBidAmount

2)  Bidding Start Time 
           ...Customers place bid on the product
               Bidding Start Time End
     Bidding duration can be set manually but for now, let it be 24 hours i.e. per day

3) After evaluating the results of all the bids say bid result, the product will be made available to the user at a price of Bid result.

** Seller and User dashBoards**

![Screenshot 2021-06-06 202102](https://user-images.githubusercontent.com/56065895/120930349-90ed4000-c70a-11eb-94ee-cf3f360be58d.png)

![Screenshot 2021-06-06 202341](https://user-images.githubusercontent.com/56065895/120930350-921e6d00-c70a-11eb-8411-c189edc6e363.png)

**Add On**

- One-to-One Bargaining
        - Chat System
        -User can directly interact to Seller
-Reminders & Notifications on New Biddings
-Users can get all Past bid result prices of a product.

**Impacts -**
Building a relationship between buyer and seller
Increased customer loyalty

Thanks

**Comment by championswimmer** [comment]
Sounds a lot like Ebay of the old days, but I love the idea for sure :) 
Should there be a time-frame within which bids need to be placed etc ?

**Comment by TAbhay** [comment]
@championswimmer **Thanks for liking the idea**
For consideration of time frame, there can be two main situations
1) Price of Products may fluctuate
2) Traffic factor
  a ) Seller with low traffic 
        In this case, no time - frame, the seller can directly interact with customers through a built-in chat system, bargain vice-versa.
  b)  Seller with high traffic 
       Large numbers of request from customers to seller which will be difficult to handle. In this case, the time frame will play a vital role 
       in deciding the price of the product.
